src/Arcus.Observability.Telemetry.Serilog.Enrichers/Extensions/LogEventPropertyValueExtensions.cs

```diff
@@ -49,15 +49,23 @@ public static double GetAsDouble(this IReadOnlyDictionary<string, LogEventProper
         /// <remarks>The built-in <c>ToString</c> wraps the string with quotes</remarks>
         /// <param name="eventPropertyValues">Event property value to provide a string representation</param>
         /// <param name="propertyKey">Key of the property to return</param>
-        public static IReadOnlyDictionary<ScalarValue, LogEventPropertyValue> GetAsDictionary(this IReadOnlyDictionary<string, LogEventPropertyValue> eventPropertyValues, string propertyKey)
+        /// <param name="throwExceptionWhenNotFound">Indication whether or not an exception should be thrown when it's not found or return null</param>
+        public static IReadOnlyDictionary<ScalarValue, LogEventPropertyValue> GetAsDictionary(this IReadOnlyDictionary<string, LogEventPropertyValue> eventPropertyValues, string propertyKey, bool throwExceptionWhenNotFound = true)
         {
```

Comment:
Can't this then be a `.TryGetAsDictionary...` with an output var, that seems the .NET approach for this. Instead of booleans controlling flow.

Reply:
Do you like this more? https://github.com/arcus-azure/arcus.observability/pull/115/commits/12477c390b6aac636427e7a3ea743fd9c1659fe0

Reply:
I like F# more 😅, but in C#, yes I think this is more natural.